src/lickey/LicenseManager.cpp

```diff
@@ -471,7 +506,7 @@ namespace lickey
         }
 
         std::ostringstream dataSection(std::ios::binary);
-        unsigned int fileVersion = VERSION();
+        char fileVersion = VERSION();
```

Comment:
this is question:
why do you change the type of fileVersion into char?

Reply:
implict conversion (or explicit) where fileVersion usage. 

here https://github.com/akio-tanaka/lickey/blob/69ba8fba88e77e99b1f3ab9d3e2ea141248ffc83/src/lickey/LicenseManager.cpp#L378

Reply:
make sense. thank you.

Reply:
Okay. I'm ready to implement next features.

Reply:
const int validLen = CalcBase64EncodedSize(32) + 8;
        if(static_cast<size_t>(validLen) > decryptedImplSize)
        {
            LOG(error) << "invalid data section";
            return false;
        }

Decryption failed here。